src/dual.jl

```diff
@@ -189,6 +189,17 @@ Base.ceil( ::Type{T}, z::Dual) where {T<:Real} = ceil( T, value(z))
 Base.trunc(::Type{T}, z::Dual) where {T<:Real} = trunc(T, value(z))
 Base.round(::Type{T}, z::Dual) where {T<:Real} = round(T, value(z))
 
+Base.zero(::Type{Dual{T}}) where {T} = Dual(zero(T), zero(T))
+Base.zero(x::Dual{T}) where {T} = zero(T)
+Base.iszero(z::Dual{T}) where {T} = iszero(value(z)) & iszero(epsilon(z))
```

Comment:
This definition of iszero is inconsistent with how `== 0` is defined for duals:
https://github.com/JuliaDiff/DualNumbers.jl/blob/52a6061b8d5e557cb8b61abda4655ccc89fa21b2/src/dual.jl#L167